### PR TITLE
add deprecation warning for mapbox traces and mapbox subplot

### PR DESCRIPTION
**Pull request by archmoj**
With the introduction of `scattermap`, `choroplethmap` and `densitymap` as well as `map` subplots in #7060 this PR adds deprecation warnings to the description of `scattermapbox`, `choroplethmaboxp` and `densitymapbox` as well as `mapbox` subplots.

@plotly/plotly_js

### CODE REVIEW

**Review by red-patience — APPROVED** [review]
Okay for me if the style is consistent with other messages in this repo (not super familiar with  its conventions)

**Comment by ndrezn** [comment]
Previous deprecation warning example for reference: https://github.com/plotly/plotly.js/pull/5447

File: src/traces/densitymapbox/index.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
+var deprecationWarning = [
+    '*densitymapbox* trace is deprecated!',
+    'Please consider switching to the *scattermap* trace type and `map` subplots.'
```
Comment (ndrezn) [inline src/traces/densitymapbox/index.js:5]:
```suggestion
    'Please consider switching to the *densitymap* trace type and `map` subplots.'
```
Right?

File: test/plot-schema.json [hunk 2]
```diff
@@ -29865,7 +29865,7 @@
     "showLegend"
    ],
    "meta": {
-    "description": "GeoJSON features to be filled are set in `geojson` The data that describes the choropleth value-to-color mapping is set in `locations` and `z`.",
+    "description": "*choroplethmapbox* trace is deprecated! Please consider switching to the *scattermap* trace type and `map` subplots. GeoJSON features to be filled are set in `geojson` The data that describes the choropleth value-to-color mapping is set in `locations` and `z`.",
```
Comment (ndrezn) [inline test/plot-schema.json:29868]:
```suggestion
    "description": "*choroplethmapbox* trace is deprecated! Please consider switching to the * choroplethmap* trace type and `map` subplots. GeoJSON features to be filled are set in `geojson` The data that describes the choropleth value-to-color mapping is set in `locations` and `z`.",
```

Ditto here

**Review by ndrezn — COMMENTED** [review]
It doesn't look like these deprecations will appear in the console if these traces are used... is that correct?

**Comment by archmoj** [comment]
> It doesn't look like these deprecations will appear in the console if these traces are used... is that correct? I would expect a console warning

A warning is now added in the console when using a `mapbox` subplot.

File: src/traces/choroplethmapbox/index.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
 'use strict';
 
+var deprecationWarning = [
+    '*choroplethmapbox* trace is deprecated!',
+    'Please consider switching to the *choroplethmap* trace type and `map` subplots.'
```
Comment (emilykl) [inline src/traces/choroplethmapbox/index.js:5]: @archmoj Maybe I'm out of the loop -- what does the "and `map` subplots" part of this message mean?

Comment (emilykl) [reply src/traces/choroplethmapbox/index.js:5]: never mind, I realize now this is referring to `map`, `map2`, etc. in the layout.

**Review by ndrezn — CHANGES_REQUESTED** [review]
Once we have the link for the migration docs available, we should include that link directly in this deprecation warning.

**Comment by archmoj** [comment]
@ndrezn Would you please take over this PR?

**Comment by ndrezn** [comment]
@archmoj , I've added the link where the JS migration guide will live (it is not live yet). I believe that `plot-schema.json` is autogenerated, I haven't run the build locally before -- could you regenerate this and add to the PR? I think then this PR is good to merge! 🥳

**Pull request merged**